frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/EditManualOptionsWithEditor/EditManualOptionsWithEditor.tsx

```diff
@@ -26,26 +31,34 @@ export function EditManualOptionsWithEditor({
     });
   };
 
+  const handleClose = () => {
+    if (component.options !== undefined) {
+      setChosenOption(true);
+    }
+
+    manualOptionsModalRef.current?.close();
+  };
+
   return (
-    <>
-      <StudioProperty.Button
-        onClick={() => manualOptionsModalRef.current.showModal()}
-        property={t('ux_editor.modal_properties_code_list_custom_list')}
-        value={buttonValue}
-      />
+    <StudioModal.Root>
+      <StudioModal.Trigger className={classes.modalTrigger} variant='secondary'>
+        {t('general.create_new')}
+      </StudioModal.Trigger>
       <StudioModal.Dialog
         ref={manualOptionsModalRef}
         className={classes.manualTabDialog}
         contentClassName={classes.content}
         closeButtonTitle={t('general.close')}
         heading={t('ux_editor.modal_add_options_code_list')}
+        onInteractOutside={handleClose}
+        onBeforeClose={handleClose}
       >
         <StudioCodeListEditor
           codeList={component.options ?? []}
-          onChange={(optionList) => handleOptionsChange(optionList)}
+          onChange={handleOptionsChange}
           texts={editorTexts}
         />
       </StudioModal.Dialog>
```

Comment:
The dialogs in `EditManualOptionsWithEditor` and `ManualOptionListEditorModal` are very similar. Can they be made into a common component, so that any future changes to the modal don't have to be updated two different places?